centrifuge-app/src/pages/Loan/FinanceForm.tsx

```diff
@@ -65,8 +66,21 @@ export const FinanceForm: React.VFC<{ loan: LoanType }> = ({ loan }) => {
   )
   const poolReserve = pool?.reserve.available.toDecimal() ?? Dec(0)
   const maxBorrow = poolReserve.lessThan(availableFinancing) ? poolReserve : availableFinancing
+  const maxRepay = balance.lessThan(loan.outstandingDebt.toDecimal()) ? balance : loan.outstandingDebt.toDecimal()
   const canRepayAll = debtWithMargin.lte(balance)
 
+  const allowedToBorrow: Record<LoanInfo['type'], boolean> = {
+    CreditLineWithMaturity:
+      'maturityDate' in loan.loanInfo &&
+      Dec(daysBetween(loan.originationDate, loan.loanInfo.maturityDate)).gt(0) &&
+      loan.outstandingDebt.toDecimal().lt(availableFinancing),
+    BulletLoan:
+      'maturityDate' in loan.loanInfo &&
+      Dec(daysBetween(loan.originationDate, loan.loanInfo.maturityDate)).gt(0) &&
+      loan.totalRepaid.toDecimal().lt(initialCeiling),
```

Comment:
I think this should be:
```suggestion
      loan.totalBorrowed.toDecimal().lt(initialCeiling),
```

Reply:
good point, just noticed that with `totalRepaid` the form would still be open even if the `maxBorrowed` is reached